federationtypes.go

```diff
@@ -0,0 +1,97 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// A SendRequest is the content of a request to PUT /_matrix/federation/v1/send/{txnID}/
+type SendRequest Transaction
+
+// A SendResponse is the content of a response to PUT /_matrix/federation/v1/send/{txnID}/
+type SendResponse struct {
+	// Map of string room event ID to the result of processing that event.
+	PDUs map[string]PDUResult `json:"pdus"`
+}
+
+// A PDUResult is the result of processing a matrix room event.
+type PDUResult struct {
+	// If not empty then this is a human readable description of a problem
+	// encountered processing an event.
+	Error string `json:"error,omitempty"`
+}
+
+// A StateIDsResponse is the content of a response to GET /_matrix/federation/v1/state_ids/{roomID}/{eventID}
+type StateIDsResponse struct {
+	// A list of state event IDs for the state of the room before the requested event.
+	StateEventIDs []string `json:"pdu_ids"`
+	// A list of event IDs needed to authenticate the state events.
+	AuthEventIDs []string `json:"auth_chain_ids"`
+}
+
+// A StateResponse is the content of a response to GET /_matrix/federation/v1/state/{roomID}/{eventID}
+type StateResponse struct {
+	// A list of events giving the state of the room before the request event.
+	StateEvents []Event `json:"pdus"`
+	// A list of events needed to authenticate the state events.
+	AuthEvents []Event `json:"auth_chain"`
+}
+
+// A MakeJoinResponse is the content of a response to GET /_matrix/federation/v1/make_join/{roomID}/{userID}
+type MakeJoinResponse struct {
+	JoinEvent Event `json:"event"`
```

Comment:
I'd like to know more about what this is. I assume it is an `m.room.member` event for the given user in the given room, but presumably this doesn't "activate" until `/send_join` is called?